lib/downloadManager.js

```diff
@@ -157,28 +167,35 @@ async function unzipPkg(zipFileDir, unzipDir, filterStr) {
 
 async function downloadSamsungCertfile() {
     console.info('[webide-common-tizentv]downloadSamsungCertfile start');
-	  let certGenFoler = 'https://developer.samsung.com/sdk-manager/repository/tizen-certificate-extension_2.0.42.zip';
-	  if (!fs.existsSync(tmpDir)) {
-		    console.info('[webide-common-tizentv]create dir:' + tmpDir);
-		    fs.mkdirSync(tmpDir);
-	  }
-
-	  let streamFile = fs.createWriteStream(certZipFile);
-	  console.info('[webide-common-tizentv]requet to download ' + certGenFoler);
-	  await request(certGenFoler).pipe(streamFile).on('close',async () => {
-		    console.info('[webide-common-tizentv]downloadSamsungCertfile(): download certificate file successful');
-		    //unzipSamsungCertifile();
+    let certGenFoler = 'https://developer.samsung.com/sdk-manager/repository/tizen-certificate-extension_2.0.42.zip';
+    if (!fs.existsSync(tmpDir)) {
+        console.info('[webide-common-tizentv]create dir:' + tmpDir);
+        fs.mkdirSync(tmpDir);
+    }
+
+    let userProxy = npmConf.get('http-proxy');
+    if (userProxy != undefined && userProxy != '') {
+        console.info('[webide-common-tizentv]downloadSamsungCertfile(): userProxy = ' + userProxy);
+        requestOpt = {'url':certGenFoler, 'proxy': userProxy};
+    } else {
+        requestOpt = certGenFoler;
+    }
+
+    let streamFile = fs.createWriteStream(certZipFile);
+    console.info('[webide-common-tizentv]requet to download ' + certGenFoler);
+    await request(requestOpt).pipe(streamFile).on('close',async () => {
+        console.info('[webide-common-tizentv]downloadSamsungCertfile(): download certificate file successful');
         await unzipPkg(tmpDir, tmpDir, 'binary');
 
         let certAddfileArr = fs.readdirSync(tmpDir);
         let certAddFileName = 'cert-add-on_2.0.42_windows-64.zip';
-    		for (let fileName of certAddfileArr) {
+        for (let fileName of certAddfileArr) {
             if (fileName.indexOf(platform) > 0) {
                 certAddFileName = fileName;
             } else {
                 fs.unlinkSync(path.resolve(tmpDir, fileName));
             }
-    		}
+        }
         await unzipPkg(tmpDir, tmpDir, 'data/tools/certificate-manager/plugins');
         await unzipPkg(tmpDir, toolsDir + path.sep + 'SamsungCertificate', 'res/ca');
```

Comment:
As I checked the final output, I checked followed 
``` bash
org.tizen.common.cert_1.0.0.201911210714\res\ca>ls
gear_test_author_CA.cer    gear_test_public_CA.cer          samsung_tizen_dev_public2.crt  vd_tizen_dev_platform2.crt
gear_test_partner_CA.cer   samsung_tizen_dev_partner2.crt   vd_tizen_dev_author_ca.cer     vd_tizen_dev_public2.crt
gear_test_platform_CA.cer  samsung_tizen_dev_platform2.crt  vd_tizen_dev_partner2.crt
```
And do you remove gear and VD files?

Reply:
These are downloaded cert files for samsung certification manager, vd files are necessary files. we just ensure samsung cert files are existed.